### ISSUE TITLE
/.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/include/absl/strings/string_view.h(495): warning: expression has no effect

### ISSUE THREAD

**Issue by novauto-nju** [Andy97/DeepMLS#2]
Hi~ Complete TensorFlow after modifying #include "cuComplex.h"  #include "cuda.h" ,I  code    bash build.sh   again.But I encountered some tough problems：

.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/include/absl/strings/string_view.h(495): warning: expression has no effect

.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/include/absl/types/optional.h(425): warning: expression has no effect
          detected during instantiation of "const T &absl::optional<T>::operator*() const & [with T=stream_executor::dnn::AlgorithmDesc]"

.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/include/tensorflow/stream_executor/dnn.h(802): here

.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/include/absl/types/optional.h(425): warning: expression has no effect
          detected during instantiation of "const T &absl::optional<T>::operator*() const & [with T=size_t]"
.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/include/tensorflow/stream_executor/dnn.h(856): here

Do you know how to solve it?

**Comment by novauto-nju** [comment]
**When I ignore the error in bash build.sh, I try Usage --
Inference using pre-trained model, I met the following alarm. Excuse me, does this affect the results of the experiment?**

**I found three new files in the Examples folder, with the suffix (.txt.txt.xyz).**

**Looking forward to your reply. Thank you！**

> INFO from ocnn: The octree key is 32 bits, the octree depth should be smaller than 8.
> 2021-05-14 00:31:22.129404: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
> decoder_octree_depth setting to 7
> using 20000 sdf samples for evaluation in each iteration
> node receptive field times: 2.0
> radius range = 2.0
> KNN=10
> utilize 1 gpus with total training batch size=32
**> WARNING:tensorflow:From DeepMLS_Generation.py:109: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.**
> 
> =====
> Program running...
> inference using prediction octree
**> WARNING:tensorflow:From /home/zenghui/deepmls/network_architecture.py:140: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.**
> 
> =======================
> input octree feature with shape:(4, ?)
> ==========================
**> WARNING:tensorflow:From Octree/ocnn-tf/libs/__init__.py:220: The name tf.get_variable is deprecated. Please use tf.compat.v1.get_variable instead.**
> 
**> WARNING:tensorflow:
> The TensorFlow contrib module will not be included in TensorFlow 2.0.**
> For more information, please see:
>   * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
>   * https://github.com/tensorflow/addons
>   * https://github.com/tensorflow/io (for I/O related ops)
> If you depend on functionality not listed there, please file an issue.
> 
**> WARNING:tensorflow:From Octree/ocnn.py:142: batch_normalization (from tensorflow.python.layers.normalization) is deprecated and will be removed in a future version.**
> Instructions for updating:
> Use keras.layers.BatchNormalization instead.  In particular, `tf.control_dependencies(tf.GraphKeys.UPDATE_OPS)` should not be used (consult the `tf.keras.layers.batch_normalization` documentation).
**> WARNING:tensorflow:From /home/zenghui/.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/python/layers/normalization.py:327: Layer.apply (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.**
> Instructions for updating:
> Please use `layer.__call__` method instead.
> input split label with shape in depth 7: (1, ?)
> input split label with shape in depth 6: (1, ?)
> input split label with shape in depth 5: (1, ?)
> input split label with shape in depth 4: (1, ?)
> input split label with shape in depth 3: (1, ?)
> input split label with shape in depth 2: (1, ?)
**> WARNING:tensorflow:From /home/zenghui/.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/python/ops/array_ops.py:1475: where (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.**
> Instructions for updating:
> Use tf.where in 2.0, which has the same broadcast rule as np.where
**> WARNING:tensorflow:From DeepMLS_Generation.py:447: The name tf.segment_sum is deprecated. Please use tf.math.segment_sum instead.**
> 
**> WARNING:tensorflow:From DeepMLS_Generation.py:505: The name tf.global_variables is deprecated. Please use tf.compat.v1.global_variables instead.**
> 
> 669 global variables
**> WARNING:tensorflow:From /home/zenghui/deepmls/utils.py:13: The name tf.trainable_variables is deprecated. Please use tf.compat.v1.trainable_variables instead.**
> 
> 1592211 model parameters total
> WARNING:tensorflow:From DeepMLS_Generation.py:512: The name tf.train.Saver is deprecated. Please use tf.compat.v1.train.Saver instead.
> 
> WARNING:tensorflow:From DeepMLS_Generation.py:517: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.
> 
> WARNING:tensorflow:From DeepMLS_Generation.py:519: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.
> 
> 2021-05-14 00:31:27.312119: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
> 2021-05-14 00:31:27.325477: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2199940000 Hz
> 2021-05-14 00:31:27.327824: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55ea5231c020 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
> 2021-05-14 00:31:27.327856: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
> 2021-05-14 00:31:27.329184: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
> 2021-05-14 00:31:27.387351: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties:
> name: PH402 SKU 200 major: 6 minor: 0 memoryClockRate(GHz): 1.0505
> pciBusID: 0000:06:00.0
> 2021-05-14 00:31:27.387445: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
> 2021-05-14 00:31:27.389512: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
> 2021-05-14 00:31:27.390884: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10.0
> 2021-05-14 00:31:27.391246: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10.0
> 2021-05-14 00:31:27.393517: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10.0
> 2021-05-14 00:31:27.394892: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10.0
> 2021-05-14 00:31:27.399570: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
> 2021-05-14 00:31:27.403490: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
> 2021-05-14 00:31:28.337076: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
> 2021-05-14 00:31:28.337125: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0
> 2021-05-14 00:31:28.337142: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N
> 2021-05-14 00:31:28.342227: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 12909 MB memory) -> physical GPU (device: 0, name: PH402 SKU 200, pci bus id: 0000:06:00.0, compute capability: 6.0)
> 2021-05-14 00:31:28.344432: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55ea52f33cc0 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
> 2021-05-14 00:31:28.344476: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): PH402 SKU 200, Compute Capability 6.0
> ===========================
> restore training from iteration 191632: Pretrained/d7_1p_pretrained/iter191631.ckpt
> ===========================
> 2021-05-14 00:31:37.197052: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
> 2021-05-14 00:31:37.881150: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
>

**Comment by novauto-nju** [comment]
**When I ignore the error in bash build.sh, I try Usage --
Inference using pre-trained model, I met the following alarm. Excuse me, does this affect the results of the experiment?**

**I found three new files in the Examples folder, with the suffix (.txt.txt.xyz).**

**Looking forward to your reply. Thank you！**

> INFO from ocnn: The octree key is 32 bits, the octree depth should be smaller than 8.
> 2021-05-14 00:31:22.129404: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
> decoder_octree_depth setting to 7
> using 20000 sdf samples for evaluation in each iteration
> node receptive field times: 2.0
> radius range = 2.0
> KNN=10
> utilize 1 gpus with total training batch size=32
**> WARNING:tensorflow:From DeepMLS_Generation.py:109: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.**
> 
> =====
> Program running...
> inference using prediction octree
**> WARNING:tensorflow:From /home/zenghui/deepmls/network_architecture.py:140: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.**
> 
> =======================
> input octree feature with shape:(4, ?)
> ==========================
**> WARNING:tensorflow:From Octree/ocnn-tf/libs/__init__.py:220: The name tf.get_variable is deprecated. Please use tf.compat.v1.get_variable instead.**
> 
**> WARNING:tensorflow:
> The TensorFlow contrib module will not be included in TensorFlow 2.0.**
> For more information, please see:
>   * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
>   * https://github.com/tensorflow/addons
>   * https://github.com/tensorflow/io (for I/O related ops)
> If you depend on functionality not listed there, please file an issue.
> 
**> WARNING:tensorflow:From Octree/ocnn.py:142: batch_normalization (from tensorflow.python.layers.normalization) is deprecated and will be removed in a future version.**
> Instructions for updating:
> Use keras.layers.BatchNormalization instead.  In particular, `tf.control_dependencies(tf.GraphKeys.UPDATE_OPS)` should not be used (consult the `tf.keras.layers.batch_normalization` documentation).
**> WARNING:tensorflow:From /home/zenghui/.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/python/layers/normalization.py:327: Layer.apply (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.**
> Instructions for updating:
> Please use `layer.__call__` method instead.
> input split label with shape in depth 7: (1, ?)
> input split label with shape in depth 6: (1, ?)
> input split label with shape in depth 5: (1, ?)
> input split label with shape in depth 4: (1, ?)
> input split label with shape in depth 3: (1, ?)
> input split label with shape in depth 2: (1, ?)
**> WARNING:tensorflow:From /home/zenghui/.conda/envs/deep_mls/lib/python3.7/site-packages/tensorflow_core/python/ops/array_ops.py:1475: where (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.**
> Instructions for updating:
> Use tf.where in 2.0, which has the same broadcast rule as np.where
**> WARNING:tensorflow:From DeepMLS_Generation.py:447: The name tf.segment_sum is deprecated. Please use tf.math.segment_sum instead.**
> 
**> WARNING:tensorflow:From DeepMLS_Generation.py:505: The name tf.global_variables is deprecated. Please use tf.compat.v1.global_variables instead.**
> 
> 669 global variables
**> WARNING:tensorflow:From /home/zenghui/deepmls/utils.py:13: The name tf.trainable_variables is deprecated. Please use tf.compat.v1.trainable_variables instead.**
> 
> 1592211 model parameters total
> WARNING:tensorflow:From DeepMLS_Generation.py:512: The name tf.train.Saver is deprecated. Please use tf.compat.v1.train.Saver instead.
> 
> WARNING:tensorflow:From DeepMLS_Generation.py:517: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.
> 
> WARNING:tensorflow:From DeepMLS_Generation.py:519: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.
> 
> 2021-05-14 00:31:27.312119: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
> 2021-05-14 00:31:27.325477: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2199940000 Hz
> 2021-05-14 00:31:27.327824: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55ea5231c020 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
> 2021-05-14 00:31:27.327856: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
> 2021-05-14 00:31:27.329184: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
> 2021-05-14 00:31:27.387351: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties:
> name: PH402 SKU 200 major: 6 minor: 0 memoryClockRate(GHz): 1.0505
> pciBusID: 0000:06:00.0
> 2021-05-14 00:31:27.387445: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
> 2021-05-14 00:31:27.389512: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
> 2021-05-14 00:31:27.390884: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10.0
> 2021-05-14 00:31:27.391246: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10.0
> 2021-05-14 00:31:27.393517: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10.0
> 2021-05-14 00:31:27.394892: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10.0
> 2021-05-14 00:31:27.399570: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
> 2021-05-14 00:31:27.403490: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
> 2021-05-14 00:31:28.337076: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
> 2021-05-14 00:31:28.337125: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0
> 2021-05-14 00:31:28.337142: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N
> 2021-05-14 00:31:28.342227: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 12909 MB memory) -> physical GPU (device: 0, name: PH402 SKU 200, pci bus id: 0000:06:00.0, compute capability: 6.0)
> 2021-05-14 00:31:28.344432: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55ea52f33cc0 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
> 2021-05-14 00:31:28.344476: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): PH402 SKU 200, Compute Capability 6.0
> ===========================
> restore training from iteration 191632: Pretrained/d7_1p_pretrained/iter191631.ckpt
> ===========================
> 2021-05-14 00:31:37.197052: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
> 2021-05-14 00:31:37.881150: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
>

**Comment by Andy97** [comment]
Hi, the .xyz file is the predicted moving least squares points, which can be opened using meshlab.
For extracting surfaces defined by these points, please follow [these steps](https://github.com/Andy97/DeepMLS#extract-isosurface-from-mls-points).
